tests/testthat/test-store.R

```diff
@@ -17,3 +17,66 @@ test_that("file store", {
   st$del(file)
   expect_false(st$exists(file))
 })
+
+test_that("caching file store", {
+  lava <- function(...) stop("Can't touch this")
+
+  filename <- tempfile()
+  bytes <- as.raw(sample(0:255, 1000, TRUE))
+  writeBin(bytes, filename)
+  h <- digest::digest(bytes, serialize = FALSE)
+
+  st <- file_store$new()
+  expect_false(st$db$exists(filename))
+
+  expect_equal(st$get_hash(filename),
+               unname(tools::md5sum(filename)))
+  expect_true(st$db$exists(filename))
+  expect_equal(st$db$get(filename)$size, 1000)
+  expect_equal(st$db$get(filename)$mtime, file.mtime(filename))
+  expect_equal(st$db$get(filename)$hash, h)
+
+  ## Does not call back to the underlying hash function:
+  with_mock("hash_files" = lava,
```

Comment:
Need to set "remake::hash_files" so that tests work in R CMD check. (Don't know why.)